packages/ui-components/src/lib/components/detail/OrderDetail.svelte

```diff
@@ -91,6 +99,64 @@
 			errToast('Failed to refresh');
 		}
 	};
+
+	const vaultTypes: { key: string; type: RaindexVaultType }[] = [
+		{ key: 'Output vaults', type: 'output' },
+		{ key: 'Input vaults', type: 'input' },
+		{ key: 'Input & output vaults', type: 'inputOutput' }
+	];
+	const getDefaultVaultsGroupedByType = (): VaultsGroupedByType => ({
+		output: [],
+		input: [],
+		inputOutput: []
+	});
+
+	$: vaultsGroupedByTypes =
+		$orderDetailQuery?.data?.vaults?.reduce((acc, vault) => {
+			const type: RaindexVaultType = vault.vaultType ?? 'inputOutput';
+			acc[type].push(vault);
+			return acc;
+		}, getDefaultVaultsGroupedByType()) || getDefaultVaultsGroupedByType();
+
+	const getWithdrawAllLabel = (type: RaindexVaultType) => {
+		switch (type) {
+			case 'output':
+				return 'Withdraw outputs';
+			case 'input':
+				return 'Withdraw inputs';
+			case 'inputOutput':
+				return 'Withdraw all';
+			default:
+				return 'Withdraw all vaults';
+		}
+	};
+
+	// It returns vaults of single group filtered by owner and non-zero balance
+	// It makes it possible to withdraw multiple vaults of the same type at once
+	// Even if some vaults of that type has zero balance
+	$: getVaultsGroupFiltered = (type: RaindexVaultType) => {
+		return vaultsGroupedByTypes[type].filter(
+			(vault) => isAddressEq($account, vault.owner) && vault.balance > 0n
+		);
+	};
+
+	// It checks does it make sense to display Withdraw button for the vault type
+	$: shouldShowWithdrawByType = (type: RaindexVaultType) => {
```

Comment:
why is this line reactive?

Reply:
`getVaultsGroupFiltered` is reactive as well, so we need to keep all of them actual and re-render if some data changed